Adds new updates for Fleet alerting rule templates (#5888)
elastic/docs-content: reference/fleet/alerting-rule-templates.md

Although these templates are managed by Elastic, any alert created from them is owned by the customer and will not be modified by Elastic, even if the templates change.

:::{important}

> Although the alerts can be used as provided, threshold values should always be evaluated in the context of your specific environment. Depending on how you adjust the thresholds, you can either generate too many alerts or fail to generate alerts when expected.

Reviewer: Reading the context here, I think this fits better. Using `can` looks like the outcome is something that users would be happy about (/actively looking for) whereas here, this is a warning: you might end up with too many alerts if you don’t adjust the threshold values:

```suggestion
Although the alerts can be used as provided, threshold values should always be evaluated in the context of your specific environment. Depending on how you adjust the thresholds, you might either generate too many alerts or fail to generate alerts when expected.
```

1. In {{kib}}, go to **{{manage-app}}** > **{{integrations}}**.
1. Find and open the integration.
1. On the integration page, select the **Alerting** tab to view all available alerting rule templates for that integration.

    :::{note}

>     The **Alerting** tab is available for all package types starting in version 9.4.0. In earlier versions, alerting rule templates are located in the **Assets** tab.

Reviewer: Based on my reasoning in this [comment](https://github.com/elastic/docs-content/pull/5888#issuecomment-4250374285):

```suggestion
    The **Alerting** tab is available for all integrations starting in version 9.4.0. In earlier versions, alerting rule templates are located in the **Assets** tab.
```

>     The **Alerting** tab is available for integration packages starting in version 9.4.0. In earlier versions, alerting rule templates are located in the **Assets** tab.

Reviewer 2: I'm aware that "integration" tends to be used to mean "package" in some contexts, but to be precise the Alerting tab is not only available for integration-type packages: https://github.com/elastic/kibana/pull/255925

    :::

    ![The Alerting tab showing available alerting rule templates for an integration](images/alerting-rule-template.png)

1. Select a template to open a prefilled **Create rule** form.

    You can use the template to create your own custom alerting rule by adjusting values, setting up connectors, and defining rule actions.

1. Review and (optionally) customize the pre-filled settings, then save and enable the rule.

>    The rule created from the template is listed on the **Rules** page. Go to **{{manage-app}}** → **Alerts and insights** → **Rules**, or access rules from solution-specific pages such as **Observability** → **Alerts** → **Manage Rules**.

Reviewer: _Alerts and insights_ is not a clickable menu, so this might be a bit misleading?

> To update a rule you created from a template, go to the **Rules** page, open the action menu (…) for the rule, and select **Edit rule**.

Reviewer: I think you can use `` {icon}`icon_name` `` and use an icon name from [this list](https://docs-v3-preview.elastic.dev/elastic/docs-builder/tree/main/syntax/icons) in the text. The closest possible would be the `boxes_vertical` icon. It’s not exactly a perfect match, but it might be better than using `(…)`?

```suggestion
To update a rule you created from a template, go to the **Rules** page, open the action menu {icon}`boxes_vertical` for the rule, and select **Edit rule**.
```

For details about fields in the Create rule form and how the rule evaluates data, refer to the [{{es}} query rule type](/explore-analyze/alerting/alerts/rule-type-es-query.md).

## Idle data streams template

```{applies_to}
stack: ga 9.4
serverless: ga
```

> Starting in version 9.4.0, all integrations include a dynamically generated **Idle data streams** template. This template generates an alert if no data is written to any of the integration's data stream patterns within a specified time period (the default is 24 hours).

Reviewer 2: This is correct. I just want to draw attention to the fact that this is only true for integration-type packages, not other types such as input.

For details about fields in the Create rule form and how the rule evaluates data, refer to the [{{es}} query rule type](/explore-analyze/alerting/alerts/rule-type-es-query.md).

## Idle data streams template

```{applies_to}
stack: ga 9.4
serverless: ga
```

> Starting in version 9.4.0, all integration-type packages include a dynamically generated **Idle data streams** template. This template generates an alert if no data is written to any of the integration's data stream patterns within a specified time period (the default is 24 hours). Note that Idle data streams are not generated for input-type packages. 

Reviewer: ```suggestion
Starting in version 9.4.0, all integrations include a dynamically generated **Idle data streams** template. This template generates an alert if no data is written to any of the integration's data stream patterns within a specified time period (the default is 24 hours). Note that Idle data streams are not generated for input-only packages. 
```

^ For this one, we had a longer conversation with Jill. The main take-aways:

> - Package is the umbrella concept; many parts of the UI and docs use integration in a broader (somewhat inaccurate) sense
> - Technically speaking, there are currently 3 types of packages: integration, input, content
> - OTel Input Package is a special type of input package
> - ECS-based vs OTel-based is a different concept than package type

Reviewer: And this was the conclusion:

> the input type (ecs, otel) [...] is not relevant to the logic:
> - The UI shows the Alerting tab for all packages
> - The Idle data streams alerting rule template is created for integration type packages only

Use this template to detect data collection issues early, such as:

- An agent or data source going offline
- Network connectivity problems preventing data ingestion
- Configuration errors stopping data flow

The Idle data streams template:

- Is named `[{Integration name}] Idle data streams`
- Appears in the **Alerting** tab alongside any bundled templates

> - Is generated automatically and isn't bundled with the integration package files

Reviewer: ```suggestion
- Is generated automatically and isn't bundled with the integration
```
